packages/entando-apimanager/modules/api/apiManager.js

```diff
@@ -169,23 +169,28 @@ export const makeRealRequest = (request, page) => {
     }
     if (response.status === 401) {
       store.dispatch(logoutUser({ statusCode: 401, request, response }));
-      throw new Error('permissionDenied');
+      throw new ErrorResponse('permissionDenied', response);
+    } else if (response.status === 400) {
+      throw new ErrorResponse('badRequest', response);
     } else if (response.status.toString().startsWith(5)) {
       if (response.status === 503) {
         store.dispatch(logoutUser({ statusCode: 401, request, response }));
-        throw new Error('serviceUnavailable');
+        throw new ErrorResponse('serviceUnavailable', response);
       } else {
-        throw new Error('serverError');
+        throw new ErrorResponse('serverError', response);
       }
     }
     return response;
   }).catch((e) => {
-    const message = `app.${normalizeErrorMessage(e.message)}`;
-    return Promise.reject(new ErrorI18n(
-      message,
-      defaultMessages[message],
-      { domain: getDomain(store.getState()) },
-    ));
+    const promise = e.response && e.response.json ? e.response.json() : Promise.resolve();
+    return promise.then((json) => {
+      const message = `app.${normalizeErrorMessage(e.message)}`;
+      const params = [message, defaultMessages[message], { domain: getDomain(store.getState()) }];
+      if (json) {
+        params.push(json.errors);
```

Comment:
why no spread operator? in this way we don't even need the if

Reply:
now I thought about it, I just removed the spread operator and instead just added the 4th parameter